DisguiseUnityRenderStream/Editor/DisguiseRenderStreamBuild.cs

```diff
@@ -2,21 +2,39 @@
 using System.Collections.Generic;
 using System.Linq;
 using UnityEditor;
+using UnityEditor.Build;
 using UnityEditor.Build.Reporting;
 using UnityEditor.SceneManagement;
+using UnityEditor.UnityLinker;
 using UnityEngine;
 using UnityEngine.Rendering;
 using UnityEngine.SceneManagement;
 using Object = UnityEngine.Object;
 
 namespace Disguise.RenderStream
 {
-    class DisguiseRenderStreamBuildPreProcessor : UnityEditor.Build.IPreprocessBuildWithReport
+    /// <summary>
+    /// Build callback order:
+    /// 1. IPreprocessBuildWithReport
+    /// 2. IUnityLinkerProcessor (only when managed code stripping is enabled, for <see cref="ReflectedMemberPreserver"/>)
+    /// 3. IPostprocessBuildWithReport
+    /// </summary>
+    class DisguiseRenderStreamBuild :
+        IPreprocessBuildWithReport,
+        IUnityLinkerProcessor,
+        IPostprocessBuildWithReport
     {
         public int callbackOrder { get; set; } = 0;
+
+        bool m_HasGeneratedSchema;
+        DisguiseRenderStreamSettings m_Settings;
+        ManagedSchema m_Schema;
+        int m_NumScenesInBuild;
         
-        public void OnPreprocessBuild(BuildReport report)
+        void IPreprocessBuildWithReport.OnPreprocessBuild(BuildReport report)
         {
+            m_HasGeneratedSchema = false;
```

Comment:
Brainstorm: So if I understand correctly, the main reason why GenerateSchema is called from either `GenerateAdditionalLinkXmlFile` or `OnPostprocessBuild` is that `GenerateAdditionalLinkXmlFile` is only called when IL2Cpp is present is that it?

If that is the case, what about we do GenerateSchema in the PreProcess like we used to do and we always fill a `ReflectedMemberPreserver` (stored in the member variables).  Then, if `GenerateAdditionalLinkXmlFile` is called we can do `GenerateAdditionalLinkXmlFile`.  

I admit, we do a little bit of processing for nothing, but it allows us to:
1. Keep the code simpler
2. (which is personally the most important one) Keep the compilation process more similar between IL2Cpp and .Net, so less chance of a different behaviour between the two cases (and less stuff we might miss when testing, or less test to do)